sources/packages/backend/apps/api/src/route-controllers/program-year/_tests_/e2e/program-year.students.controller.getProgramYearById.e2e-spec.ts

```diff
@@ -0,0 +1,120 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { ProgramYear } from "@sims/sims-db";
+import * as request from "supertest";
+import { DataSource } from "typeorm";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getStudentToken,
+  FakeStudentUsersTypes,
+} from "../../../../testHelpers";
+
+describe("ProgramYearStudentsController(e2e)-getProgramYearById", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let programYear2001: ProgramYear;
+  let programYear2026: ProgramYear;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    appDataSource = dataSource;
+    const programYearRepo = appDataSource.getRepository(ProgramYear);
+
+    programYear2001 = {
```

Comment:
For E2E tests, the idea is to avoid the pain of individually creating objects and setting all their properties, that is why we have, for instance, the `createFakeSomething` factories. The idea is to generate dummy values and set only the ones needed for testing.
We do have a `createFakeProgramYear` that can be, for instance, adapted to receive a year and generate all other fake properties based on that parameter.

For instance, lines 23 to 46 can be converted to:
```ts
const programYear2001Promise = programYearRepo.save(createFakeProgramYear(2001));
const programYear2026Promise = programYearRepo.save(createFakeProgramYear(2026));
```